docs/cunumeric/source/user/notebooks/black_scholes.ipynb

```diff
@@ -0,0 +1,233 @@
+    "\\frac{1}{2\\pi}\\int_{-\\infty}^{x} e^{-t^2/2}\\; dt = \\frac12\\big[1 + \\mathtt{erf}\\big(\\frac{x}{\\sqrt{2}}\\big)\\big]\n",
+    "$$\n",
+    "\n",
+    "In order to compute this efficiently, an expansion approximation (from John C. Hull (1997) “Options, Futures, and Other Derivatives”) is used."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 90,
+   "id": "b798400e-b726-46cf-83de-cbe0e613737f",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def CND(d):\n",
+    "    # The expansion coefficients for erf are derived from Abramowitz and Stegun, eq 7.1.26 \n",
+    "    # https://www.cs.bham.ac.uk/~aps/research/projects/as/resources/AandS-a4-v1-2.pdf\n",
+    "    A1 = 0.31938153\n",
+    "    A2 = -0.356563782\n",
+    "    A3 = 1.781477937\n",
+    "    A4 = -1.821255978\n",
+    "    A5 = 1.330274429\n",
+    "    RSQRT2PI = 0.39894228040143267793994605993438  # 1 / sqrt(2*pi)\n",
+    "\n",
+    "    K = 1.0 / (1.0 + 0.2316419 * np.abs(d))\n",
+    "\n",
+    "    cnd = RSQRT2PI * np.exp(-0.5*d*d) * (K*(A1 + K*(A2 + K*(A3 + K*(A4 + K*A5)))))\n",
+    "\n",
+    "    return np.where(d > 0, 1.0 - cnd, cnd)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8e4ae70e-30ac-47cf-949d-444695b5cacc",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Now that we have a method to compute the CND, we can imlement the Black-Scholes formula below"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 91,
+   "id": "fc601408-39db-4149-afd6-2866eab73f8f",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def black_scholes(S, X, T, r, v):\n",
+    "    sqrt_T = np.sqrt(T)\n",
+    "    \n",
+    "    d1 = np.log(S / X) + (r + 0.5 * v * v) * T / (v * sqrt_T)\n",
+    "    cnd_d1 = CND(d1)\n",
+    "\n",
+    "    d2 = d1 - v * sqrt_T\n",
+    "    cnd_d2 = CND(d2)\n",
+    "    \n",
+    "    exp_rT = np.exp(-r * T)\n",
+    "    \n",
+    "    call_result = S * cnd_d1 - X * exp_rT * cnd_d2\n",
+    "    put_result = X * exp_rT * (1.0 - cnd_d2) - S * (1.0 - cnd_d1)\n",
+    "\n",
+    "    return call_result, put_result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "31b2da3e-1c67-4f07-a485-77320ec0210e",
+   "metadata": {},
+   "source": [
+    "Let's use the `black_scholes` function above to price one put and call million options, keeping everying fixed except the current option price $S$. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 92,
+   "id": "d539c451-6c1b-4afd-a09f-86c70d7990e9",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Elapsed Time: 57.517 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "from legate.timing import time\n",
+    "\n",
+    "N = 1000000                 # number of options to price\n",
+    "S = np.linspace(5, 100, N)  # current option price\n",
+    "X = 100                     # strike price\n",
```

Comment:
These examples are usually much more compelling if you can compare CPU vs. multi-GPU timings.  If we can get the resource-scoping-pr merged, we could compute this twice with a different machine and show GPU times being way smaller.

Reply:
That's not going to be possible in a single notebook, is it? AFAIK the Realm runtime has to be configured at startup and can't be changed. I'm also not sure we can assume users who want to run the notebooks will have multi-gpus. cc @manopapad 

I do think comparisons would be great to have, but I think we should do them, and write up the results with some plots, in a ReST chapter.

Reply:
I do think we could include information in the notebooks about how to re-run them with a different configuration (but I do think they would have to be exited, and re-run)

Reply:
With the resource-scoping-pr, we could do the following:

Cell #1
```
def function_that_does_everything(...):

```

Cell #2:
```
with machine.only("cpu"):
  function_that_does_everything()
```

Cell #3:
```
with machine.only("gpu"):
  function_that_does_everything()
```

Users without a GPU would just not run cell #3. Alternatively, cell #3 should fail gracefully without completely crashing the kernel if there are no GPUs. This might make more sense for a follow-up PR. Seeing GPU times that are shorter than CPU times definitely hit the reward centers of user's brains and get people engaged.

Reply:
> With the resource-scoping-pr, 

What PR are you referring to? Is it something already merged, or soon to be?

> Seeing GPU times that are shorter than CPU times definitely hit the reward centers of user's brains and get people engaged.

Of course, but we need to make sure these examples actually demonstrate that. The GPU tests often run more slowly for me locally, presumably because they are small and some startup time dominates.

Reply:
This will go in a follow-up PR once resource scoping is merged